security_checks.groovy

```diff
@@ -0,0 +1,54 @@
+
+@Library('PipelineUtils@master')
+import dvsa.aws.mot.jenkins.pipeline.common.RepoFunctions
+import dvsa.aws.mot.jenkins.pipeline.common.GlobalValues
+
+// global variables
+RepoFunctions repoFunctionsFactory = new RepoFunctions()
+GlobalValues globalValuesFactory = new GlobalValues()
+Map<String, Map<String, String>> github = globalValuesFactory.GITHUB_REPOS()
+
+String jenkinsBuildLabel = "cvr"
+String branch = "feature-BL-8921-security-checks" // TODO: change to "master" before merging
+String account = "dvsarecallsdev"
+String recallsDir = 'recalls'
+
+pipeline {
+  agent none
+  options {
+    ansiColor('xterm')
+    timestamps()
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(
+      logRotator(
+        daysToKeepStr: '7'
+      )
+    )
+    disableConcurrentBuilds()
+  }
+  stages {
+    stage('Check NPM dependencies') {
+      failFast true
+      agent { node { label "${jenkinsBuildLabel} && ${account}" } }
+      steps {
+        script {
+          dir(recallsDir) {
+            if (!repoFunctionsFactory.checkoutGitRepo(
+              github.cvr_app.url,
+              branch,
+              github.cvr_app.name,
+              globalValuesFactory.SSH_DEPLOY_GIT_CREDS_ID
+            )) {
+              failure("Failed to clone repository ${github.cvr_app.url}; branch: ${branch}")
+            }
+            dir('recalls-app') {
+              sh 'npm prune'
+              sh 'npm run install:all'
+              sh 'npm run retire'
+            }
```

Comment:
If this will fail, there will be a big log with "No failure causes found" at the end. Maybe catch it somehow with "failure" ?

Reply:
Done